### PR TITLE
Chore(next): Include react specific eslint v9 rules

### PR DESCRIPTION
**Pull request by ahennr**
… and cleanup some `babel` plugins that are already included in `preset-env`.

Plz review @terrestris/devs

### CODE REVIEW

File: eslint.config.mjs [hunk 1]
```diff
@@ -14,28 +17,47 @@ export default tsEslint.config({
     importPlugin.flatConfigs.recommended
   ],
   files: [
-    '**/*.ts'
+    '**/*.{ts,tsx}',
```
Comment (dnlkoch) [inline eslint.config.mjs:20]:
```suggestion
    '**/*.{ts,tsx}'
```

File: src/Component/FormField/Permission/RolePermissionGrid/RolePermissionGrid.tsx [hunk 2]
```diff
@@ -57,7 +56,7 @@ const RolePermissionGrid: React.FC<RolePermissionGridProps> = ({
   };
 
   const getRoles = async (pageOpts?: PageOpts) => {
-    return await client?.role().findAll(pageOpts);
+    return client?.role().findAll(pageOpts);
```
Comment (dnlkoch) [inline src/Component/FormField/Permission/RolePermissionGrid/RolePermissionGrid.tsx:59]: Are we sure this is correct?

**Review by dnlkoch — APPROVED** [review]
Very nice, thanks @ahennr!

File: src/Component/FormField/Permission/RolePermissionGrid/RolePermissionGrid.tsx (hunk 2 above)
Comment (ahennr) [reply src/Component/FormField/Permission/RolePermissionGrid/RolePermissionGrid.tsx]: Nice catch! Reverted in 9b7243d

**Pull request merged**